controllers/authrealm/dex.go

```diff
@@ -415,16 +430,23 @@ func (r *AuthRealmReconciler) createDexConnectors(authRealm *identitatemv1alpha1
 						Namespace: authRealm.Namespace,
 					},
 					UserSearch: dexoperatorv1alpha1.UserSearchSpec{
-						BaseDN:    authRealm.Spec.LDAPExtraConfigs[idp.Name].BaseDN,
-						Filter:    authRealm.Spec.LDAPExtraConfigs[idp.Name].Filter,
-						Username:  idp.LDAP.Attributes.PreferredUsername[0],
+						BaseDN:    url.BaseDN,
+						Filter:    url.Filter,
+						Scope:     scope,
+						Username:  url.QueryAttribute,
 						IDAttr:    idp.LDAP.Attributes.ID[0],
 						EmailAttr: idp.LDAP.Attributes.Email[0],
 						NameAttr:  idp.LDAP.Attributes.Name[0],
 					},
 					GroupSearch: authRealm.Spec.LDAPExtraConfigs[idp.Name].GroupSearch,
 				},
 			}
+			if url.Scheme == "ldap" {
```

Comment:
Is this logic correct or backwards?  Wouldn't ldaps want TLS?

Reply:
Its other way round. if there ldaps its using TLS connection to server. If there ldap, then wwe have to providee startTLS =true, to upgrade the connection to TLS

Reply:
<img width="702" alt="Screen Shot 2022-05-18 at 8 25 17 AM" src="https://user-images.githubusercontent.com/58599706/169038149-62f07aaa-020e-4fa9-bd91-38ce3365543c.png">